paddle/fluid/operators/recurrent_op.cc

```diff
@@ -377,16 +377,35 @@ void RecurrentGradOp::RunImpl(const framework::Scope &scope,
           zero_op->Run(scope, place);
         }
 
-        auto new_inside_name = cur_scope.Rename(inside_grad_name);
-
-        // sum gradient
-        auto sum_op = framework::OpRegistry::CreateOp(
-            "sum", {{"X", {pg_names[param_id], new_inside_name}}},
-            {{"Out", {pg_names[param_id]}}},
-            framework::AttributeMap{{"use_mkldnn", {false}}});
-        sum_op->Run(cur_scope, place);
-
-        cur_scope.Rename(new_inside_name, inside_grad_name);
+        // add inside grad variable to outside grad variable
+        // if subblock's variable affect parent block varieble.
+        if (inside_grad_name.compare(pg_names[param_id]) != 0) {
```

Comment:
Why?

Reply:
find the variables have same name in both block so they can share data